src/Swift.Bindings/src/Parser/ModuleProcessor.cs

```diff
@@ -0,0 +1,223 @@
+        /// <param name="module">The name of the Swift module being processed.</param>
+        /// <param name="typeDecls">Dictionary of Swift type specs to their declarations.</param>
+        /// <param name="dylibPath">File path to the Swift dynamic library.</param>
+        public ModuleProcessor(string module, string dylibPath,
+            Dictionary<NamedTypeSpec, TypeDecl> typeDecls, TypeDatabase typeDatabase)
+        {
+            _module = module;
+            _dylibPath = dylibPath;
+            _typeDatabase = typeDatabase;
+            _typeDecls = typeDecls;
+        }
+
+        /// <summary>
+        /// Executes the post-processing workflow for all unprocessed types in this module.
+        /// Produces type database entries for structs, enums, and classes.
+        /// </summary>
+        public void FinalizeTypeProcessing()
+        {
+            foreach (var (typeSpec, typeDecl) in _typeDecls)
+            {
+                if (!_typeDatabase.IsTypeProcessed(typeSpec.Module, typeSpec.NameWithoutModule))
+                {
+                    ProcessTypeRecursively(typeSpec, typeDecl);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Recursively processes a type. If the type is a struct, enum, or class,
+        /// calls into specialized handlers.
+        /// </summary>
+        /// <param name="namedTypeSpec">The Swift type specification.</param>
+        /// <param name="typeDecl">The associated type declaration.</param>
+        private void ProcessTypeRecursively(NamedTypeSpec namedTypeSpec, TypeDecl typeDecl)
+        {
+            if (_typeDatabase.IsTypeProcessed(namedTypeSpec.Module, namedTypeSpec.NameWithoutModule))
+                return;
+
+            switch (typeDecl)
+            {
+                case StructDecl structDecl:
+                    ProcessStruct(namedTypeSpec, structDecl);
+                    break;
+
+                case EnumDecl enumDecl:
+                    ProcessEnum(namedTypeSpec, enumDecl);
+                    break;
+
+                case ClassDecl classDecl:
+                    ProcessClass(namedTypeSpec, classDecl);
+                    break;
+
+                default:
+                    throw new NotSupportedException($"Unknown or unsupported type declaration '{typeDecl.GetType().Name}'.");
+            }
+        }
+
+        /// <summary>
+        /// Processes a Swift struct declaration to determine final properties such as blittability
+        /// and frozenness, then registers it into the type database.
+        /// </summary>
+        /// <param name="namedTypeSpec">Swift type specification for the struct.</param>
+        /// <param name="structDecl">The struct declaration.</param>
+        private void ProcessStruct(NamedTypeSpec namedTypeSpec, StructDecl structDecl)
+        {
+            // Ensure all fields are processed or known in the database.
```

Comment:
All nominal types can have inner types.  Because fields in the structure may depend on them, we should process any inner types before we process fields.

Reply:
They are processed during ABI parsing so we have full information. Added test for that and patched some logic.